homeassistant/components/keyboard_remote.py

```diff
@@ -0,0 +1,122 @@
+"""
+Recieve signals from a keyboard and use it as a remote control.
+
+This component allows to use a keyboard as remote control. It will
+fire ´keyboard_remote_command_received´ events witch can then be used
+in automation rules.
+
+The `evdev` package is used to interface with the keyboard and thus this
+is Linux only. It also means you can't use your normal keyboard for this,
+because `evdev` will block it.
+
+Example:
+  keyboard_remote:
+    device_descriptor: '/dev/input/by-id/foo'
+    key_value: 'key_up' # optional alternaive 'key_down' and 'key_hold'
+    # be carefull, 'key_hold' fires a lot of events
+
+  and an automation rule to bring breath live into it.
+
+  automation:
+    alias: Keyboard All light on
+    trigger:
+      platform: event
+      event_type: keyboard_remote_command_received
+      event_data:
+        key_code: 107 # inspect log to obtain desired keycode
+    action:
+      service: light.turn_on
+      entity_id: light.all
+"""
+
+# pylint: disable=import-error
+import threading
+import logging
+
+from homeassistant.const import (
+    EVENT_HOMEASSISTANT_START,
+    EVENT_HOMEASSISTANT_STOP
+)
+
+DOMAIN = "keyboard_remote"
+REQUIREMENTS = ['evdev>=0.6.1']
+_LOGGER = logging.getLogger(__name__)
+ICON = 'mdi:remote'
+KEYBOARD_REMOTE_COMMAND_RECEIVED = 'keyboard_remote_command_received'
+KEY_CODE = 'key_code'
+KEY_VALUE = {'key_up': 0, 'key_down': 1, 'key_hold': 2}
+TYPE = 'type'
+DEVICE_DESCRIPTOR = 'device_descriptor'
+
```

Comment:
Can you please add configuration validation with `voluptuous`? We are migrating all existing components/platforms at the moment.